enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,288 @@
+
+The user will boot a machine with this liveCD, the liveCD will start executing a similar flow as a bootstrap node in a regular installation:
+bootkube.sh will do all the rendering logic.
+The live ISO environment provides a scratch place to write bootstrapping files so that they don't end up on the real node. This eliminates a potential source of errors and confusion when debugging problems.
+The bootstrap static pods should be generated in a way that the control plane operators will be able to identify them and either continue in a controlled way for the next revision, or just keep them as the correct revision and reuse them.
+
+cluster-bootstrap will apply all the required manifests (under /opt/openshift/manifests/)
+
+Bootkube will get the master Ignition from machine-config-server and
+enrich the master Ignition with control plane static pods manifests and all required resources including etcd data.
+Bootkube.sh will write the master Ignition along with rhcos to disk.
+At this point bootkube will reboot the node and let it finish the cluster creation.
+
+Post reboot:
+The kubelet service will start the control plane static pods.
+Kubelet will send a CSR (see below) and join the cluster.
+CVO will deploy all cluster operators.
+The control plane operators will rollout a new revision (if necessary).  
+
+#### OpenShift-installer
+
+We will add logic to the installer to create `bootstrap-in-place-for-live-iso.ign` Ignition config.
+This Ignition config will diverge from the bootstrap Ignition:
+bootkube.sh:
+1. Start cluster-bootstrap without required pods (`--required-pods=''`)
+2. Run cluster-bootstrap with `--bootstrap-in-place` entrypoint to enrich the master Ignition.
+3. Write RHCOS image and the master Ignition to disk.
+4. Reboot the node.
+
+#### Cluster-bootstrap
+
+By default, `cluster-bootstrap` start the bootstrap control plane and create all the manifests under /opt/openshift/manifests.
+`cluster-bootstrap` also wait for a list required pods to be ready, these pods are expetced to start running on the master nodes.
+In case we are running the bootstrap in place, there is no master node that can run the pods. cluster-bootstrap should apply the manifest and tear down the control plane, in case it fail to apply some of the manifest it should return an error.
+
+
+`cluster-bootstrap` will have a new entrypoint `--bootstrap-in-place` which will get the master Ignition as input and will enrich the master Ignition with control plane static pods manifests and all required resources including etcd data.
+
+#### Bootstrap / Control plane static pods
+The control plane components we will add to the master Ignition are (to be placed under `/etc/kubernetes/manifests`):
+1. etcd-pod
+2. kube-apiserver-pod
+3. kube-controller-manager-pod
+4. kube-scheduler-pod
+
+Control plane required resources to be added to the Ignition:
+1. `/var/lib/etcd`
```

Comment:
Hopefully this will always be small, but are there any limits to the size of files that Ignition can deploy that may come into play here?

Reply:
According to @cgwalters there is no size limit.